Sources/SwiftProtobuf/_MessageStorage+BinaryEncoding.swift

```diff
@@ -494,4 +492,55 @@ extension _MessageStorage {
             encode(value, &encoder)
         }
     }
+
+    /// Serializes the field tag and values for a repeated (packed or unpacked) `enum` field.
+    private func serializeRepeatedEnumField(
+        for fieldNumber: Int,
+        field: FieldLayout,
+        into encoder: inout BinaryEncoder,
+        isPacked: Bool
+    ) throws {
+        if isPacked {
+            // First, iterate over the values to compute the packed length.
+            var length = 0
+            _ = try layout.performOnRawEnumValues(
+                _MessageLayout.TrampolineToken(index: field.submessageIndex),
+                field,
+                self,
+                .read
+            ) {
+                length += Varint.encodedSize(of: $0)
+                return true
+            } /*onInvalidValue*/ _: { _ in
+            }
+
+            encoder.startField(fieldNumber: fieldNumber, wireFormat: .lengthDelimited)
+            encoder.putVarInt(value: length)
+
+            // Then, iterate over them again to encode the actual varints.
+            _ = try layout.performOnRawEnumValues(
+                _MessageLayout.TrampolineToken(index: field.submessageIndex),
+                field,
+                self,
+                .read
+            ) {
+                encoder.putVarInt(value: Int64($0))
+                return true
+            } /*onInvalidValue*/ _: { _ in
+            }
+        } else {
+            // Iterate over the raw values and encode each as its own tag and varint.
+            _ = try layout.performOnRawEnumValues(
+                _MessageLayout.TrampolineToken(index: field.submessageIndex),
+                field,
+                self,
+                .read
+            ) {
+                encoder.startField(fieldNumber: fieldNumber, wireFormat: .varint)
+                encoder.putVarInt(value: Int64($0))
```

Comment:
Given one truncates on reading enums, we could probably do a 32bit version that doesn't have to sign extend for these, but it means we'd need to do the sizing also correctly do realize they will be shorter.

Reply:
Longer-term, I'd like to turn `BinaryEncoder` into a purely low-level API that would be the opposite of `WireFormatReader` (`WireFormatWriter`), but that needs to wait until I've removed all the binary visitor code.